core/src/main/scala/spinal/core/internals/PhaseVerilog.scala

```diff
@@ -96,8 +99,10 @@ class PhaseVerilog(pc: PhaseContext, report: SpinalReport[_]) extends PhaseMisc
       () => componentBuilderVerilog.result
     } else {
       emitedComponentRef.put(component, oldComponent)
+      val comments = "\n" + component.rtlComments.map(_.split("\n")).flatten.map(comment => f"$commentSymbol $comment").mkString("\n") + "\n"
+      val str      =  s"$commentSymbol ${component.definitionName} replaced by ${oldComponent.definitionName}\n"
       component.definitionName = oldComponent.definitionName
-      () => s"\n//${component.definitionName} replaced by ${oldComponent.definitionName}\n"
```

Comment:
Are you fixing a hidden bug here? Seems that previously, `component.definitionName` was overwritten before use. Now the string is built before overwriting which makes sense. Or, if this wasn’t a bug, perhaps it is one now?

Reply:
Good question.. I have no idea.. it is weird because in the PhaseVerilog and PhaseVhdl it is not the same (https://github.com/SpinalHDL/SpinalHDL/blob/dev/core/src/main/scala/spinal/core/internals/PhaseVhdl.scala#L81, https://github.com/SpinalHDL/SpinalHDL/blob/dev/core/src/main/scala/spinal/core/internals/PhaseVerilog.scala#L99). @Dolu1990 which version is correct ?

Reply:
Hooo nice catch XD

Yes that was a bug ^^